### PR TITLE
532 update market cards

### PR DESCRIPTION
**Pull request by robhyrk**
- updated styles based on new designs
- created reusable HorizontalScroll component
- adjusted logic for cards shown based on container width
- added dynamic widths for cards based on container width

note:
- some styles may not be accurately shown due to new breakpoints and the left/right drawers still in place
- kept old code temporarily in /market-card/_index.tsx and ./markets/_MarketScroll.tsx in case we need to reference/revert back to it (will remove these once index page is approved)
- will also cleanup remaining commented code once index page is approved

### CODE REVIEW

File: components/markets/MarketScroll.tsx [hunk 1]
```diff
@@ -1,38 +1,48 @@
...
       "right",
     );
     const { width: containerWidth, ref: containerRef } = useResizeDetector();
-    const cardWidth = 320;
-    const gap = 30;
+    const gap = 16;
+    //calculate cards shown and width based on container width
+    const cardsShown =
+      containerWidth < 716
+        ? 3
```
Comment (TvrtkoM) [inline components/markets/MarketScroll.tsx:27]: Think this is a typo. on small screens it shows only one card but scrolls 3 at a time.

File: components/markets/MarketScroll.tsx [hunk 2]
```diff
@@ -1,38 +1,48 @@
...
+        : containerWidth >= 716 && containerWidth < 1183
+        ? 2
+        : 3;
+    // const cardsShown = Math.floor(containerWidth / (gap + cardWidth));
```
Comment (TvrtkoM) [inline components/markets/MarketScroll.tsx:31]: can this be removed?

File: components/markets/MarketScroll.tsx [hunk 3]
```diff
@@ -1,38 +1,48 @@
...
     const scrollMin = 0;
     const scrollMax = cardWidth * markets.length + gap * (markets.length - 1);
-    const cardsShown = Math.floor(containerWidth / (gap + cardWidth));
+
     const moveSize = cardsShown * (cardWidth + gap);
 
     useEffect(() => {
       scrollRef.current.scroll({ left: scrollLeft, behavior: "smooth" });
-    }, [scrollRef, scrollLeft]);
+    }, [scrollRef, scrollLeft, cardWidth]);
```
Comment (TvrtkoM) [inline components/markets/MarketScroll.tsx:45]: strange behavior in my opinion, when the screen width (and the cardWidth) is changed, the component scrolls the market cards. On smallest breakpoint cards can end up looking like this
image

File: components/markets/MarketScroll.tsx [hunk 4]
```diff
@@ -59,58 +69,36 @@ const MarketScroll = observer(
...
-            </button>
-          </div>
-        </div>
-        <div className="relative">
+      <div ref={containerRef} className="grid sm:grid-cols-2 gap-4 md:gap-6">
+        <h3 className="sm:col-span-1 font-bold text-[28px]">{title}</h3>
+        <HorizontalScroll
+          classes="order-2 sm:order-none"
+          link={link}
+          cta="Go To Markets"
```
Comment (TvrtkoM) [inline components/markets/MarketScroll.tsx:77]: Think `cta` should also be passed as a prop to the component. Maybe use a specific type for both `cta` and `link`.

File: pages/markets/index.tsx [hunk 5]
```diff
@@ -35,7 +35,7 @@ const MarketsPage: NextPage<{
           <MarketScroll
             title="Newest Markets"
             markets={newestMarkets}
-            showMarketsLink={false}
+            // showMarketsLink={false}
```
Comment (TvrtkoM) [inline pages/markets/index.tsx:38]: think this comment can be removed now.

**Comment by TvrtkoM** [comment]
there is a console error now when going to `markets/` route
```
Warning: `NaN` is an invalid value for the `minWidth` css style property.
```

**Review by TvrtkoM — CHANGES_REQUESTED** [review]
Looks good. Think it needs some adjustments. Refer to comments above

**Comment by lsaether** [comment]
Can you work with @mk26071980 to define hover behavior on the market row scroll component? Something similar to what was implemented on the new learn cards.

File: components/markets/MarketScroll.tsx (hunk 1 above)
Comment (robhyrk) [reply components/markets/MarketScroll.tsx:27]: I made a mistake on the scroll functionality breakpoint; scrolling shouldn't appear until 905px breakpoint. I have made the adjustment now.

The market cards should stack one-by-one until hitting 905px so the logic on cardsShown for < 716 is unnecessary; I have removed it.

File: components/markets/MarketScroll.tsx (hunk 2 above)
Comment (robhyrk) [reply components/markets/MarketScroll.tsx:31]: yes

File: components/markets/MarketScroll.tsx (hunk 3 above)
Comment (robhyrk) [reply components/markets/MarketScroll.tsx:45]: Yes, actually upon review adding cardWidth to the useEffect is unnecessary. Width prop gets updating just fine without it.

**Comment by robhyrk** [comment]
Thanks for the feedback. I made the changes and added the hover states as per [designs](https://www.figma.com/file/A0OEqjIWy6jL4Rk4kT8ddr/UX%2FUI_Workflow?node-id=1327%3A8329&t=elNlcY7q4Vs6cUHk-0)

**Review by TvrtkoM — APPROVED** [review]

**Pull request merged**